libs/networking/http_helpers/http_helpers.ml

```diff
@@ -35,176 +43,115 @@ let client_ref : (module Cohttp_lwt.S.Client) option ref = ref None
 let in_mock_context = ref false
 let set_client_ref v = if not !in_mock_context then client_ref := Some v
 
+let with_client_ref v f x =
+  let old = !client_ref in
+  set_client_ref v;
+  let result = f x in
+  (match old with
+  | Some old -> set_client_ref old
+  | None -> client_ref := None);
+  result
```

Comment:
Not obvious to me why we don't set the ref when `in_mock_context`. Should probably be documented in the mli if it is relevant to the caller too.

If we don't need this, then also consider `Common.save_excursion` (related: I think that function should be renamed for clarity).

Reply:
cc @aryx who's name is on the blame. How come we have `in_mock_context`?